File: src/main/java/org/raku/comma/actions/ExtractRegexAction.java
package org.raku.comma.actions;

import com.intellij.lang.refactoring.RefactoringSupportProvider;
import com.intellij.refactoring.RefactoringActionHandler;
import com.intellij.refactoring.actions.BasePlatformRefactoringAction;
import org.raku.comma.refactoring.RakuExtractRegexPartHandler;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ExtractRegexAction extends BasePlatformRefactoringAction {
    @Nullable
    @Override
    protected RefactoringActionHandler getRefactoringHandler(@NotNull RefactoringSupportProvider provider) {
        return new RakuExtractRegexPartHandler();
    }

    @Override
    protected boolean isAvailableInEditorOnly() {
        return true;
    }
}

File: src/main/java/org/raku/comma/actions/ExtractRegexAction.kt
package org.raku.comma.actions

import com.intellij.lang.refactoring.RefactoringSupportProvider
import com.intellij.refactoring.RefactoringActionHandler
import com.intellij.refactoring.actions.BasePlatformRefactoringAction
import org.raku.comma.refactoring.RakuExtractRegexPartHandler

class ExtractRegexAction : BasePlatformRefactoringAction() {
    override fun getRefactoringHandler(provider: RefactoringSupportProvider): RefactoringActionHandler? {
        return RakuExtractRegexPartHandler()
    }

    override fun isAvailableInEditorOnly(): Boolean = true
}
